src/matchers/numbers.test.lua

```diff
@@ -0,0 +1,653 @@
+						normalNumber
+					)
+				)
+
+				assertMatcher.hasMessage(
+					resultSwapped,
+					("Expected inf to be close to %d within 1 digits"):format(
+						normalNumber
+					)
+				)
+			end)
+
+			test("on NaN for any parameter", function()
+				-- Given
+				local normalNumber = 1
+
+				-- When
+				local result =
+					matchers.toBeCloseTo(CONTEXT, normalNumber, NAN, 1)
+
+				local resultSwapped =
+					matchers.toBeCloseTo(CONTEXT, NAN, normalNumber, 1)
+
+				-- Then
+				assertMatcher.failed(result)
+				assertMatcher.hasMessage(
+					result,
+					("Expected %d to be close to %s within 1 digits"):format(
+						normalNumber,
+						tostring(NAN)
+					)
+				)
+				assertMatcher.failed(resultSwapped)
+				assertMatcher.hasMessage(
+					resultSwapped,
+					("Expected %s to be close to %d within 1 digits"):format(
+						tostring(NAN),
+						normalNumber
+					)
+				)
+			end)
+		end)
+
+		it("should have an inverted message", function()
+			-- Given
+			local numDigits = 1
+
+			-- When
+			local result = matchers.toBeCloseTo(
+				INVERTED_CONTEXT,
+				closeValue,
+				testValue,
+				numDigits
+			)
+
+			-- Then
+			assertMatcher.hasMessage(
+				result,
+				("Expected %.3f to not be close to %d within 1 digits"):format(
+					closeValue,
+					testValue,
+					numDigits
+				)
+			)
+		end)
+	end)
+
+	describe("toBeGreaterThan", function()
+		it(
+			"should pass when the received value is greater than the expected value",
+			function()
+				-- Given
+				local greaterNumber = 10
+				local smallerNumber = 1
+
+				-- When
+				local result = matchers.toBeGreaterThan(
+					CONTEXT,
+					greaterNumber,
+					smallerNumber
+				)
+
+				-- Then
+				assertMatcher.passed(result)
+			end
+		)
+
+		describe("should fail", function()
+			test(
+				"when the received value is less than the expected value",
+				function()
+					-- Given
+					local greaterNumber = 10
+					local smallerNumber = 5
+
+					-- When
+					local result = matchers.toBeGreaterThan(
+						CONTEXT,
+						smallerNumber,
+						greaterNumber
+					)
+
+					-- Then
+					assertMatcher.failed(result)
+					assertMatcher.hasMessage(
+						result,
+						("Expected %d to be greater than %d"):format(
+							smallerNumber,
+							greaterNumber
+						)
+					)
+				end
+			)
+
+			test("on NaN for any parameter", function()
+				-- Given
+				local normalNumber = 1
+
+				-- When
+				local result =
+					matchers.toBeGreaterThan(CONTEXT, normalNumber, NAN)
+
+				local resultSwapped =
+					matchers.toBeGreaterThan(CONTEXT, NAN, normalNumber)
+
+				-- Then
+				assertMatcher.failed(result)
+				assertMatcher.hasMessage(
+					result,
+					("Expected %d to be greater than %s"):format(
+						normalNumber,
+						tostring(NAN)
+					)
+				)
+				assertMatcher.failed(resultSwapped)
+				assertMatcher.hasMessage(
+					resultSwapped,
+					("Expected %s to be greater than %d"):format(
+						tostring(NAN),
+						normalNumber
+					)
+				)
+			end)
```

Comment:
For basic operations like greater than or less than, we don't need to assert the behaviour ourselves. The matcher will always behave the same as Lua because it's written in Lua using the built in operators.

Reply:
I meant you don't need to assert how it handles NaN or infinity specifically as that's not affecting the logic.

Instead you deleted all the tests...